dali/pipeline/operators/python_function/dltensor_function.h

```diff
@@ -106,13 +106,15 @@ class DLTensorPythonFunctionImpl : public PythonFunctionImplBase<Backend> {
     py::gil_scoped_acquire interpreter_guard{};
     py::object output_o;
     try {
+      if (std::is_same<Backend, GPUBackend>::value) cudaDeviceSynchronize();
```

Comment:
It may be overkill. Maybe we should put some constraint on the user about using streams.
Like provide him stream he should work on, and if decide to use his own it is up to him to synchronize before and after (any only stream provided by DALI and one he works on)?

Reply:
After some consideration I think we should:
- provide an operator argument that will tel if DALI stream should be synchronized before calling python function. So here is should be cudaStreamSynchornize (to make sure that the data is already copied to the memory that op will use)
- provide the user a stream number he can work on - it is up to him if he can/want to schedule work on the DALI stream and skip/manage synchronization on his own
- the synchronization after python function should be removed and up to the user

To sum up:

| Operator should synchronize DALI stream before calling python function| User uses DALI stream| User needs to synchronize at the end of the python code|
|----------|---------------|-------|
| no |  yes | no|
| yes | no | yes |

Reply:
Now:
It's *always* user's responsibility to synchronize work done within the operator, although they can use the stream provided by `current_dali_stream()` function. 
DALI synchronizes its stream before the operator call when the `synchronize_stream` flag is set to true (it is by default).